### ISSUE TITLE
Commander Core (PID 0c32): System.IO.IOException: Operation failed early

### ISSUE THREAD

**Issue by EvanMulawski** [EvanMulawski/FanControl.CorsairLink#6]
> Unfortunately it's not working for me. Here's a bit of logging:

```
20-1-2023 14:48:51: [CorsairLink] DeviceManager: Found 3 Corsair device(s)
  name=LCD Cap for Elite Capellix coolers, devicePath=\\?\hid#vid_1b1c&pid_0c39#7&2544036e&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_01#8&2a7d46db&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}

20-1-2023 14:48:51: [CorsairLink] DeviceManager: Found 1 supported Corsair device(s)
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}

20-1-2023 14:48:51: [CorsairLink] An exception occurred attempting to initialize device 'CORSAIR iCUE COMMANDER Core (205537824242)' (\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}):
20-1-2023 14:48:51: [CorsairLink] System.IO.IOException: Operation failed early: The parameter is incorrect ---> System.ComponentModel.Win32Exception: The parameter is incorrect
   --- End of inner exception stack trace ---
   at HidSharp.Platform.Windows.NativeMethods.OverlappedOperation(IntPtr ioHandle, IntPtr eventHandle, Int32 eventTimeout, IntPtr closeEventHandle, Boolean overlapResult, NativeOverlapped* overlapped, UInt32& bytesTransferred)
   at HidSharp.Platform.Windows.WinHidStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at CorsairLink.CommanderCoreDevice.SendCommand(ReadOnlySpan`1 command, ReadOnlySpan`1 data)
   at CorsairLink.CommanderCoreDevice.Initialize()
   at CorsairLink.CommanderCoreDevice.Connect()
   at FanControl.CorsairLink.CorsairLinkPlugin.FanControl.Plugins.IPlugin.Initialize()
```

AC V143, and the beta-4 release
It's the commander that came with the H170i and one that came with the 7000d case.

Maybe I'm doing something wrong? If you need more info, feel free to msg me. Thanks for making this!

_Originally posted by @bjongbloed in https://github.com/EvanMulawski/FanControl.CorsairLink/discussions/1#discussioncomment-4737697_

**Comment by EvanMulawski** [comment]
Hi @bjongbloed. This is an issue specific to the `0c32` variant of the Commander CORE (a.k.a. Commander ST). This is an extremely new controller that is shipping with new AIO's, so I'm probably going to need to send you some custom plugin builds to test communication with this device.

**Comment by EvanMulawski** [comment]
@bjongbloed

1. Can you make sure your firmware is up to date and post the firmware version reported by iCUE? If there is an update available, please update and then re-test the beta.4 version of the plugin.

2. Can you post a picture of your Commander hub (if possible)?

**Comment by bjongbloed** [comment]
Thanks for the quick response. I guess I'll install their shitty software again ;-) 
![20230120_191926](https://user-images.githubusercontent.com/18465729/213776960-0b80f87b-c954-4f55-8789-7e66a478d322.JPG)
![20230120_191851](https://user-images.githubusercontent.com/18465729/213776964-b105958c-1418-4405-9add-8ef9ae27ddc8.JPG)

**Comment by bjongbloed** [comment]
I don't think that first one is a commander, am I right?

**Comment by bjongbloed** [comment]
![image](https://user-images.githubusercontent.com/18465729/213778622-2194c22f-8452-4904-a6e1-53542a0c691e.png)
No updates available.

**Comment by bjongbloed** [comment]
Yes for the pump and yes it's detachable.

Theres a seperate single wire for rpm reporting to the cpu fan connector on the mobo.

**Comment by bjongbloed** [comment]
Everything is working beautifully now. Nice!

```
21-1-2023 09:31:10: [CorsairLink] DeviceManager: Found 3 Corsair device(s)
  name=LCD Cap for Elite Capellix coolers, devicePath=\\?\hid#vid_1b1c&pid_0c39#7&2544036e&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_01#8&2a7d46db&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}

21-1-2023 09:31:10: [CorsairLink] DeviceManager: Found 1 supported Corsair device(s)
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}

21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedSensor/0
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedSensor/1
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedSensor/2
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedSensor/3
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedSensor/4
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedSensor/5
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedSensor/6
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedController/0
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedController/1
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedController/2
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedController/3
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedController/4
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedController/5
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/SpeedController/6
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/TemperatureSensor/0
21-1-2023 09:31:10: [CorsairLink] Added CorsairLink/\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}/TemperatureSensor/1
```

Only thing is my fan Leds are now off, but that's because it's in software mode now I guess. Maybe I'll have to try your new friends software :p

**Comment by EvanMulawski** [comment]
Fixed in **v1.0.0-beta.5**.
